src/raft/replication.c

```diff
@@ -1674,6 +1674,50 @@ static void takeSnapshotClose(struct raft *r, struct raft_snapshot *s)
 	r->fsm->snapshot_finalize(r->fsm, &s->bufs, &s->n_bufs);
 }
 
+
+static unsigned dynamicTrailingIndex(struct raft *r, struct raft_snapshot *snapshot) {
+	struct raft_log *l = r->log;
+	size_t snapshot_size = 0, num_entries = logNumEntries(l);
+	unsigned trailing, i;
+	/** 
+	 * This should never happen as it would mean that the snapshot
+	 * contains no new entry (i.e. either the snapshot is empty
+	 * or equal to the previous one).
+	 */
+	assert(num_entries > 0);
+
+	for (i = 0; i < snapshot->n_bufs; i++) {
+		snapshot_size += snapshot->bufs[i].len;
+	}
+
+	for (trailing = 0; trailing < num_entries; trailing++) {
+		const struct raft_entry* entry = logEntryAt(l, num_entries - trailing - 1);
+		if (entry->buf.len > snapshot_size) {
+			break;
+		}
+		snapshot_size -= entry->buf.len;
+	}
+
+	if (trailing < r->snapshot.threshold) {
+		trailing = r->snapshot.threshold;
+	} else if (trailing > r->snapshot.trailing) {
+		trailing = r->snapshot.trailing;
+	}
+
+	return trailing;
+}
+
+
+static unsigned trailingSize(struct raft *r, struct raft_snapshot *snapshot) {
```

Comment:
Could you think what's needed to make this function to take just `const`-s:
`static unsigned trailingSize(const struct raft *r, const struct raft_snapshot *snapshot)`
?
In case if it's hard please or I'm missing something please tell me.

Reply:
I added this in a separate commit (a145249) as the const propagatest (nicely) as usual in 6 other methods (and some more `static` ones, but I don't really care about those)